snap/quota/resources.go

```diff
@@ -146,27 +194,27 @@ func (qr *Resources) Change(newLimits Resources) error {
 		return err
 	}
 
+	// perform the changes initially on a dry-run so we can validate
+	// the resulting quotas combined.
+	resultingLimits := *qr
```

Comment:
This is an interesting approach, I think it works fine with the current way we have setup `Resource` and what `Change` is doing. However it makes me a bit uneasy because it may break in subtle ways in the future. The risk I see is that `resultingLimits` is not a deep copy of grp. So if we every manipulate `resultingLimits.<pointer>.<value>` instead of replacing the entire (sub)struct we will leak changes into `grp` without realizing it. Unfortunately I don't have a great idea for an alternative. Maybe a strong comment here is enough? Would love to hear the thoughts of the others too (and maybe I'm wrong)

Reply:
So this was exactly my concern with how I copy, I left it like this because I wanted to discuss it, thanks for the input so far

Reply:
I will change this to a more explicit approach to be more correct. Thanks for your review!